pkg/core/resources/model/resource.go

```diff
@@ -507,32 +534,27 @@ func ComputeLabels(
 		}
 	}
 
-	if isK8s && IsLocallyOriginated(mode, labels) {
-		ns, ok := resourceNameExtensions[mesh_proto.KubeNamespaceTag]
-		if ok && ns != "" {
-			setIfNotExist(mesh_proto.KubeNamespaceTag, ns)
-		}
+	if ns.value != "" && isK8s && IsLocallyOriginated(mode, labels) {
```

Comment:
There's some magic here that ns == UnknownNamespace would be easier to understand

Reply:
this line is fine, there is another line below `if ns.system || ns.value == "" {` that I changed now to `if ns.system || ns == UnsetNamespace`